### PR TITLE
Getting Started Rules

### LINKED ISSUE #1808: Rules and Conclusion to Getting Started
This is an issue to help track and discussion the completion of the Getting Started Tutorial. 

I'm going to have just a little bit of extra time in the coming weeks and I'd like to get what is on the forum submitted and the conclusion pages written. 

- [x] Rules: Introduction
- [x] Basic UI Rules
- [x] Blockly
- [x] Advanced Coding
- [x] Tips and Tricks and Conclusion

With the conclusion the Getting Started will be 16 pages long. Would it be the right time to consider adding at least one more level of hierarchy so on the side bar we just have the large topics which opens a page where we can navigate to the Basic/Intermediate/Advanced pages instead of the flat hierarchy which takes up a huge amount of room on the sidebar? Or would that best be handled in a separate issue?

I've no idea how that would work technically.

~~Also, would it be better to submit each of the above pages as a separate PR or wait until I get them all transferred over and submitted in one go?~~
EDIT: It went faster than I anticipated. I'll submit it all as one PR and if necessary back it out and create separate PRs.

### PR DESCRIPTION
**Pull request by rkoshak**
Added the rules pages and a tips and tricks pages to finish out the Getting Started Tutorial.

Closes #1808 

Signed-off-by: Richard Koshak <rlkoshak@gmail.com>

### CODE REVIEW

**Comment by rkoshak** [comment]
This PR should be ready for review for now.

**Review by Confectrician — COMMENTED** [review]
Is *blockly-add-comment.jpg* intended to be jpg instead of png?
Because all other pics and the most in general in the docs are in png file format.

**Review by Confectrician — COMMENTED** [review]
image

Images look good so far. I will come back to the contents and the preview this week.
Sorry for the delay.

File: tutorials/getting_started/rules_introduction.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+| Script   | rule consisting of a single Script Action and has the tag "Script", there are no triggers so they must be run manually or called from another rule | A great place to store examples when you figure something out and want to remember how you did it later. Can include reusable code called from other rules (though there are often better choices). Very useful for adhoc and pseudo-unit testing of your rules. Despite being a rule, Scripts do not appear in the Rules page of MainUI. |
+| Schedule | any rule with a time based trigger and the tag "Schedule"                                                                                          | This page is a good place to see when your rules are scheduled to run. However, note that it only shows statically timed rules.                                                                                                                                                                                                           |
+
+Important note on the term "Script".
+Unfortunately this term is overloaded in opneHAB and has multiple meanings based on the context.
```
Comment (Confectrician) [inline tutorials/getting_started/rules_introduction.md:128]:
```suggestion
Unfortunately this term is overloaded in openHAB and has multiple meanings based on the context.
```

File: tutorials/getting_started/rules_basic.md [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+
+Very basic rules can be defined in the UI without the need for code.
+These rules will follow a `when X event occurred if Y is true do Z` format.
+For example, when a member of the Temperatures Group changes, if it changed to a value over 70 °F, command the AlertLight Item to "red".
+
+For this example we will be creating two rules, one that sets a Switch Item to `ON` when the cloudiness reading goes above 50% and another to `OFF` when it falls below 50%.
+
+### Create the Rule
+
+Navigate to Settings and Rules and click on the + icon in the lower right corner.
```
Comment (Confectrician) [inline tutorials/getting_started/rules_basic.md:28]:
```suggestion
Navigate to Settings and Rules and click on the **+** icon in the lower right corner.
```

Mabye make this bold for a better understanding.

File: tutorials/getting_started/rules_basic.md [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+### Updates to Templates
+
+Your rule, once instantiated becomes its own thing separated from the template, just as if you had cut and paste the code as in the first example.
+Therefore, it will not change even if the rule template is changed or removed.
+This means the rule will not update based on changes to the template automatically.
+To update a template and a rule use the following process.
+
+1. Notice that there has been an update to a template that you want to use (keep checking back).
+1. Select the template in MainUI and click "Remove". Then click "Add" to refresh the template with the new changes.
+1. Go to Settings and Rules and navigate to the rule you want to update. Click the pause icon to disable that rule. Or if you don't want to keep a backup, delete it. Note, if you just disable you won't be able to use the same rule UID for the new rule.
```
Comment (Confectrician) [inline tutorials/getting_started/rules_basic.md:186]: ordered lists can handle one sentence per line writing too, so you could split this up and have a look at the preview.

File: tutorials/getting_started/tips-and-tricks.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+## Don't Wait, Try it Out!
+
+If you know what you want to try but you just don't know what will happen, don't post a new thread on the forum or spend hours reading through the docs and the forum.
+Instead, try it out and see what happens.
+If you have backups, there is little you could do that will actually break things.
+
+## Developer Sidebar
+
+MainUI has a tool called the Developer Sidebar.
+Bring up the sidebar through Developer Tools or the key combination <kbd>Alt-Shift-D</kbd>.
```
Comment (Confectrician) [inline tutorials/getting_started/tips-and-tricks.md:35]: I think it looks better as three single keys on the page.

```suggestion
Bring up the sidebar through Developer Tools or the key combination <kbd>Alt</kbd><kbd>Shift</kbd><kbd>D</kbd>.
```

**Review by Confectrician — COMMENTED** [review]
Just one small suggestion left.

Anything left from your sid @rkoshak ?
Otherwise i would approve and merge, when we discussed/solved the last comment. :)

File: tutorials/getting_started/tips-and-tricks.md (hunk 4 above)
Comment (rkoshak) [reply tutorials/getting_started/tips-and-tricks.md:35]: That looks good to me. I'll accept this change and commit one more update to clear out those pesky MD linting errors I introduced on the last commit and we can call this done. Woohoo!

**Comment by rkoshak** [comment]
OK, ready to merge. There is still one linting error but it's not in a file I modified so I didn't fix it.

**Review by Confectrician — APPROVED** [review]
Thanks a lot @rkoshak . 👍🏼

Again a huge step forward for the docs.
Should we reintroduce this also for the stable docs homepage?

Stable should be 3.2 currently so many users should already have most of the functions available when using stable.

**Comment by rkoshak** [comment]
I'm pretty certain there isn't anything in this that includes something past stable 3.2 unless the Blockly docs didn't get added until after 3.2. It looks like they are there so I think it will work. It would be a help to those on the stable release for sure.

**Pull request merged**